pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/withscript/WithScriptDescriptor.java

```diff
@@ -104,15 +101,17 @@ public T newInstance() throws Exception {
 
     @Extension
     public static class WithScriptAllowlist extends GroovySourceFileAllowlist {
-        private final Set<String> scriptUrls = new HashSet<>();
-
-        private void registerScript(URL scriptUrl) {
-            scriptUrls.add(scriptUrl.toString());
-        }
-
         @Override
         public boolean isAllowed(String groovyResourceUrl) {
-            return scriptUrls.contains(groovyResourceUrl);
+            for (WithScriptDescriptor descriptor : ExtensionList.lookup(WithScriptDescriptor.class)) {
+                URL scriptResource = descriptor.getScriptResource();
```

Comment:
This will be slower than before. I suspect it would be ok as-is, but https://github.com/jenkinsci/workflow-cps-plugin/pull/829 will cache all of the allowlist responses, which should make performance a non-issue.